pkg/gitops/flux/client.go

```diff
@@ -0,0 +1,116 @@
+package flux
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/config"
+	"github.com/aws/eks-anywhere/pkg/executables"
+	"github.com/aws/eks-anywhere/pkg/retrier"
+	"github.com/aws/eks-anywhere/pkg/types"
+)
+
+const (
+	maxRetries    = 5
+	backOffPeriod = 5 * time.Second
+)
+
+// FluxClient is an interface that abstracts the basic commands of flux executable.
+type FluxClient interface {
+	BootstrapGithub(ctx context.Context, cluster *types.Cluster, fluxConfig *v1alpha1.FluxConfig) error
+	BootstrapGit(ctx context.Context, cluster *types.Cluster, fluxConfig *v1alpha1.FluxConfig, cliConfig *config.CliConfig) error
+	Uninstall(ctx context.Context, cluster *types.Cluster, fluxConfig *v1alpha1.FluxConfig) error
+	SuspendKustomization(ctx context.Context, cluster *types.Cluster, fluxConfig *v1alpha1.FluxConfig) error
+	ResumeKustomization(ctx context.Context, cluster *types.Cluster, fluxConfig *v1alpha1.FluxConfig) error
+	Reconcile(ctx context.Context, cluster *types.Cluster, fluxConfig *v1alpha1.FluxConfig) error
+}
+
+// KubeClient is an interface that abstracts the basic commands of kubectl executable.
+type KubeClient interface {
+	UpdateAnnotation(ctx context.Context, resourceType, objectName string, annotations map[string]string, opts ...executables.KubectlOpt) error
+	DeleteSecret(ctx context.Context, managementCluster *types.Cluster, secretName, namespace string) error
+}
+
+type fluxClient struct {
+	flux FluxClient
+	kube KubeClient
+	*retrier.Retrier
+}
+
+func newFluxClient(flux FluxClient, kube KubeClient) *fluxClient {
+	return &fluxClient{
+		flux:    flux,
+		kube:    kube,
+		Retrier: retrier.NewWithMaxRetries(maxRetries, backOffPeriod),
+	}
+}
+
+func (c *fluxClient) BootstrapGithub(ctx context.Context, cluster *types.Cluster, fluxConfig *v1alpha1.FluxConfig) error {
+	return c.Retry(
+		func() error {
+			return c.flux.BootstrapGithub(ctx, cluster, fluxConfig)
```

Comment:
not for this pr, just a thought: it would be good to understand the specific circumstances in which these commands (the bootstrap and uninstall cmds) fail, and if there are cases where we should and shouldn't retry; temporary loss of connectivity between the admin machine and the cluster leading to failure to call the API, for example, is a good case for a retry. But I wonder if there are other cases outside of this that we're accounting for here which should just fail and not be retried. Ideally those would be caught earlier in the validations tho...